dev/sync_schema_result.h

```diff
@@ -39,6 +39,11 @@ namespace sqlite_orm {
          *      4. data_type mismatch between table and storage.
          */
         dropped_and_recreated,
+
+        /*
+         * old table is dropped and new recreated but no effort at keeping data is made
```

Comment:
not sure that sync schema result my have a dedicated value about data status. Data is lost if result is dropped and recreated with `preserve = false`

Reply:
no, if we wanted that then we could se the new enum value to express that! But I don't see the point.. however I can implement that if you insist
just let my migration API as it is because it works!!